csvjoin/functions.py

```diff
@@ -0,0 +1,257 @@
+    header = header.split(',')
+    n = len(header)
+    filler = ''
+    for _ in range(n - 1):
+        filler += 'NULL'
+        filler += ','
+    filler += 'NULL'
+
+    counter = 1
+    while line != '':
+        if counter in taken_rows:
+            line = file.readline()
+            counter += 1
+            continue
+
+        if join_type == 'left':
+            parse_data_line(line, filler, -1, column_index)
+
+        if join_type == 'right':
+            parse_data_line(filler, line, column_index, -1)
+
+        line = file.readline()
+        counter += 1
+
+
+def join_files(file_path1: str, file_path2: str, column_name: str, join_type: str):
+    """
+        The actual main algorithm for printing the joined files.
+
+        :param file_path1: The path of the first file.
+        :param file_path2: The path of the second file.
+        :param column_name: The name of the column to join on.
+        :param join_type: The type of the join.
+        :type file_path1: str
+        :type file_path2: str
+        :type column_name: str
+        :type join_type: str
+    """
+    try:
+        # Even though I can have left or right join
+        # firstly I need to do inner and after that
+        # I can do the rest if needed
+        with open(file_path1) as file1, open(file_path2) as file2:
+            header1 = str(file1.readline())
+            header2 = str(file2.readline())
+
+            # look for the index where column_name appears in header of the CSV files
+            column_index1 = find_index_of_column(header1, column_name)
+            column_index2 = find_index_of_column(header2, column_name)
+
+            if join_type == 'right':
+                parse_data_line(header1, header2, column_index1, -1)
+            else:
+                parse_data_line(header1, header2, -1, column_index2)
+
+            taken_rows1 = set()
+            taken_rows2 = set()
+
+            i1 = 1
+            line1 = file1.readline()
+            while line1 != '':
+
+                # return to the begging of the file2
+                file2.seek(0, 0)
+                i2 = 0
+                line2 = file2.readline()
+
+                while line2 != '':
+                    if i2 == 0:
+                        i2 += 1
+                        line2 = file2.readline()
+                        continue
+
+                    arr1 = split_and_strip(line1)
+                    arr2 = split_and_strip(line2)
+
+                    if arr1[column_index1] == arr2[column_index2]:
+                        taken_rows1.add(i1)
+                        taken_rows2.add(i2)
+                        if join_type == 'right':
+                            parse_data_line(line1, line2, column_index1, -1)
+                        else:
+                            parse_data_line(line1, line2, -1, column_index2)
+
+                    i2 += 1
+                    line2 = file2.readline()
+
+                i1 += 1
+                line1 = file1.readline()
```

Comment:
This is really below the main loop. Using CSV reader would make this much more readable.